File: docs/WidgetMill/flutter/sdui_flutter/lib/src/builders/text.dart
import 'package:flutter/widgets.dart';
import '../model/sdui_node.dart';
import '../parsing/enums.dart';
import '../parsing/parsers.dart';
import '../renderer.dart';

/// Text — spec §2.5.
Widget buildText(BuildContext context, SduiNode node, SduiRenderer r) {
  final p = node.props;
  return Text(
    (p['data'] ?? '').toString(),
    style: parseTextStyle(p['style']),
    textAlign: textAlignFrom(p['textAlign']),
    maxLines: (p['maxLines'] as num?)?.toInt(),
    overflow: textOverflowFrom(p['overflow']),
    softWrap: p['softWrap'] as bool? ?? true,
  );
}

File: docs/WidgetMill/packages/spec/src/nodes/text.ts
import { z } from "zod";
import { TextStyle } from "../complex/textStyle";
import { TEXT_ALIGN, TEXT_OVERFLOW } from "../enums";
import { BindableString } from "../scalars";

/** Text — spec §2.5. */
export const TextNode = z.object({
  type: z.literal("text"),
  props: z
    .object({
      data: BindableString.optional(),
      style: TextStyle.optional(),
      textAlign: z.enum(TEXT_ALIGN).optional(),
      maxLines: z.number().int().optional(),
      overflow: z.enum(TEXT_OVERFLOW).optional(),
      softWrap: z.boolean().optional(),
    })
    .default({}),
});
